modules/openapi-generator/src/main/resources/python/api_client.mustache

```diff
@@ -169,6 +170,9 @@ class ApiClient(object):
             post_params = self.parameters_to_tuples(post_params,
                                                     collection_formats)
             post_params.extend(self.files_parameters(files))
+            if header_params['Content-Type'].startswith("multipart"):
+                post_params = self.parameters_to_multipart(post_params,
+                                                          (dict) )
```

Comment:
How about:
```
if header_params['Content-Type'].startswith("multipart"):
    post_params = self.parameters_to_multipart(post_params, (dict))
else:
    post_params = self.parameters_to_tuples(post_params, collection_formats)
```
That way we don't run parameters_to_tuples when we don't need it.

Reply:
@spacether, Your changes may not work. for example,
non dict property with content-type:multipart/* will be almost passthroughed  parameters_to_multipart(just became tuple) and not converted by parameters_to_tuples() style. 
i.e There is no csv handling in parameters_to_multipart but  parameters_to_tuples has it.

Reply:
So we need them to always be tuples as inputs in parameters_to_multipart?
If so, please update the docstring there to correctly describe that params will always be tuples and will never be a dict.

Reply:
@spacether, do you mean I need to change docstring more in addition to the first coment? I think the change by first comment seems enough, but if you feel something lacked, I will add it in docstring.

Reply:
Thanks. I am all set here. I wanted the change that you made:
```
    def parameters_to_multipart(self, params, collection_types):
        """Get parameters as list of tuples, formatting as json if value is collection_types

        :param params: Parameters as list of two-tuple
```